### ISSUE TITLE
Panic "attempted to leave type `linked_hash_map::Node<u32, rusttype::gpu_cache::Row>` uninitialized, which is invalid"

### ISSUE THREAD

**Issue by rhys-vdw** [amethyst/space_shooter_rs#100]
I'm super new to Rust so let me know if there's more info required. I have just cloned and run `cargo run`, but the program panics before a window opens.

Version: 3276c8e81de6397adf399ba56b3a4686ae43015e
OS: Windows 10
rustc 1.48.0 (7eac88abb 2020-11-16)
cargo 1.48.0 (65cbdd2dc 2020-10-14)

```
thread '<unnamed>' panicked at 'attempted to leave type `linked_hash_map::Node<u32, rusttype::gpu_cache::Row>` uninitialized, which is invalid', C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\mem\mod.rs:658:9
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\panicking.rs:483
   1: core::panicking::panic_fmt
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\core\src\panicking.rs:85
   2: core::panicking::panic
             at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\core\src\panicking.rs:50
   3: core::mem::uninitialized
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\mem\mod.rs:658
   4: linked_hash_map::LinkedHashMap<u32, rusttype::gpu_cache::Row, core::hash::BuildHasherDefault<rustc_hash::FxHasher>>::ensure_guard_node<u32,rusttype::gpu_cache::Row,core::hash::BuildHasherDefault<rustc_hash::FxHasher>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\linked-hash-map-0.5.2\src\lib.rs:174
   5: linked_hash_map::LinkedHashMap<u32, rusttype::gpu_cache::Row, core::hash::BuildHasherDefault<rustc_hash::FxHasher>>::insert<u32,rusttype::gpu_cache::Row,core::hash::BuildHasherDefault<rustc_hash::FxHasher>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\linked-hash-map-0.5.2\src\lib.rs:304
   6: rusttype::gpu_cache::Cache::cache_queued<closure-7>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rusttype-0.8.3\src\gpu_cache.rs:778
   7: glyph_brush::glyph_brush::GlyphBrush<tuple<u32, amethyst_ui::pass::UiArgs>, twox_hash::std_support::sixty_four::RandomXxHashBuilder64>::process_queued<tuple<u32, amethyst_ui::pass::UiArgs>,twox_hash::std_support::sixty_four::RandomXxHashBuilder64,closure-
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\glyph_brush-0.6.3\src\glyph_brush.rs:336
   8: amethyst_ui::glyphs::{{impl}}::run<gfx_backend_vulkan::Backend>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\amethyst_ui-0.15.3\src\glyphs.rs:368
   9: shred::system::{{impl}}::run_now<amethyst_ui::glyphs::UiGlyphsSystem<gfx_backend_vulkan::Backend>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\shred-0.10.2\src\system.rs:140
  10: shred::dispatch::stage::{{impl}}::execute::{{closure}}
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\shred-0.10.2\src\dispatch\stage.rs:111
  11: core::ops::function::impls::{{impl}}::call_mut<tuple<mut arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>*>,closure-0>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:247
  12: core::slice::iter::{{impl}}::for_each<arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>,closure-0*>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\slice\iter\macros.rs:203     
  13: rayon::iter::for_each::{{impl}}::consume_iter<closure-0,mut arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>*,core::slice::iter::IterMut<arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-1.3.0\src\iter\for_each.rs:55
  14: rayon::iter::plumbing::Producer::fold_with<rayon::slice::IterMutProducer<arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>>,rayon::iter::for_each::ForEachConsumer<closure-0>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-1.3.0\src\iter\plumbing\mod.rs:110
  15: rayon::iter::plumbing::bridge_producer_consumer::helper<rayon::slice::IterMutProducer<arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>>,rayon::iter::for_each::ForEachConsumer<closure-0>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-1.3.0\src\iter\plumbing\mod.rs:438
  16: rayon::iter::plumbing::bridge_producer_consumer::helper::{{closure}}<rayon::slice::IterMutProducer<arrayvec::ArrayVec<[alloc::boxed::Box<RunNow>; 5]>>,rayon::iter::for_each::ForEachConsumer<closure-0>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-1.3.0\src\iter\plumbing\mod.rs:427
  17: rayon_core::join::join_context::call_b::{{closure}}<tuple<>,closure-1>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\join\mod.rs:130
  18: rayon_core::job::{{impl}}::execute::call::{{closure}}<tuple<>,closure-0>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\job.rs:113
  19: std::panic::{{impl}}::call_once<tuple<>,closure-0>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:308
  20: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:381
  21: std::sync::rwlock::{{impl}}::deref_mut<core::option::Option<alloc::sync::Arc<rayon_core::thread_pool::ThreadPool>>>
  22: std::panicking::try<tuple<>,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:345
  23: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\rhysv\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:382
  24: rayon_core::unwind::halt_unwinding<closure-0,tuple<>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\unwind.rs:17
  25: rayon_core::job::{{impl}}::execute<rayon_core::latch::SpinLatch,closure-0,tuple<>>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\job.rs:119
  26: rayon_core::job::JobRef::execute
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\job.rs:59
  27: rayon_core::registry::WorkerThread::execute
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:681
  28: rayon_core::registry::WorkerThread::wait_until_cold<rayon_core::latch::CountLatch>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:665
  29: rayon_core::registry::WorkerThread::wait_until<rayon_core::latch::CountLatch>
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:639
  30: rayon_core::registry::main_loop
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:759
  31: rayon_core::registry::ThreadBuilder::run
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:56
  32: rayon_core::registry::{{impl}}::spawn::{{closure}}
             at C:\Users\rhysv\.cargo\registry\src\github.com-1ecc6299db9ec823\rayon-core-1.7.0\src\registry.rs:101
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by rhys-vdw** [comment]
Ah, turns out this is already fixed on `develop` at a747c98c66f91d854ccca5d9bb1d244919a69a10

**Comment by cdsupina** [comment]
I'm glad you got working. Feel free to let us know if you have any more questions. I hope you find this showcase game helpful!
